src/05-derivatives/02-common-data-types.md

```diff
@@ -62,9 +78,54 @@ makes them invalid (e.g., if a source 4D image is averaged to create a single
 static volume, a SamplingFrequency property would no longer be relevant). In
 addition, all processed files include the following metadata JSON fields:
 
-| **Key name**  | **Description**                                                                                 |
-| ------------- | ----------------------------------------------------------------------------------------------- |
-| SkullStripped | REQUIRED. Boolean. Whether the volume was skull stripped (non-brain voxels set to zero) or not. |
+| **Key name**  | **Description**                                                                                                              |
+| ------------- | ---------------------------------------------------------------------------------------------------------------------------- |
+| SkullStripped | REQUIRED. Boolean. Whether the volume was skull stripped (non-brain voxels set to zero) or not.                              |
+| Resolution    | REQUIRED if `res` keyword present. Dictionary of Strings to Strings. Specifies the interpretation of the resolution keyword. |
+| Density       | REQUIRED if `den` keyword present. Dictionary of Strings to Strings. Specifies the interpretation of the density keyword.    |
+
+Example JSON file corresponding to
+`pipeline1/sub-001/func/sub-001_task-rest_run-1_space-MNI305_bold.json` above:
+
+```JSON
+{
+  "SkullStripped": true,
+  "Resolution": {
+    "hi": "Matched with high-resolution T1w (0.7mm, isotropic)",
+    "lo": "Matched with original BOLD resolution (2x2x3 mm^3)"
+  }
+}
+```
+
+Example of CIFTI files (i.e., a file that combines regularly sampled data
+and non-parametric surfaces) having both `res` and `den` keywords:
+
+```Text
+pipeline1/
+    sub-001/
+        func/
+            sub-001_task-rest_run-1_res-1_den-10k_bold.dtseries.nii
+            sub-001_task-rest_run-1_res-1_den-41k_bold.dtseries.nii
+            sub-001_task-rest_run-1_res-2_den-10k_bold.dtseries.nii
+            sub-001_task-rest_run-1_res-2_den-41k_bold.dtseries.nii
+            sub-001_task-rest_run-1_bold.dtseries.json
+```
+
+And the corresponding `sub-001_task-rest_run-1_bold.dtseries.json` file:
+
+```JSON
+{
+    "SkullStripped": true,
+    "Resolution": {
+        "1": "Matched with MNI152NLin6Asym 1.6mm isotropic",
+        "2": "Matched with MNI152NLin6Asym 2.0mm isotropic"
+    },
+    "Density": {
+        "10k": "10242 vertices per hemisphere (5th order icosahedron)",
+        "41k": "40962 vertices per hemisphere (6th order icosahedron)"
```

Comment:
just like these say `10k` and `41k` someone could have used `1p6` and `2p0` for `Resolution` as well, right? or conversely i could have used `1` and `2` for `Density`.

btw, this may be the first use of a dictionary as a value in bids. i know this was discussed a long time back in the context of pybids, and in our derivatives meeting to stay away from, so may be worth a little ping to bep leads and others.

Reply:
> just like these say 10k and 41k someone could have used 1p6 and 2p0 for Resolution as well, right? or conversely i could have used 1 and 2 for Density.
Sure

> btw, this may be the first use of a dictionary as a value in bids. i know this was discussed a long time back in the context of pybids, and in our derivatives meeting to stay away from, so may be worth a little ping to bep leads and others.

I just added a suggestion from @effigies that would provide an alternative to dicts. However, it seems useful to have all resolutions in just one place.